doc/code_style_guidelines.md

```diff
@@ -0,0 +1,218 @@
+
+Code style guidelines
+=====================
+
+This is the coding style used for C/C++ code. Also see the Lua code style guidelines.
+
+The coding style is based on the <https://www.kernel.org/doc/html/latest/process/coding-style.html Linux kernel code style>. Much of the existing code doesn't follow the current code style guidelines, do not try to replicate that. Use your best judgment for C++-specific syntax.
+
+Currently, the code uses C++14. Do not use features that depend on more recent versions.
+
+
+# Spelling
+
+Use American English, but avoid idioms that may be difficult to understand by non-native speakers.
+
+
+# Function declarations
+
+In case your function parameters don't fit within the defined line length, use the following style.
+Indention for continuation lines is **exactly** two tabs.
+
+```cpp
+void some_function_name(type1 param1, type2 param2, type3 param3,
+		type4 param4, type5 param5, type6 param6, type7 param7)
+{
+	...
+}
+```
+
+Sometimes with complex function declarations, it might be messy to define as many parameters as possible on the same line.
+This is acceptable too (and currently used in some places):
+
+```cpp
+void some_function_name(
+		const ReallyBigLongTypeName &param1,
+		ReallyBigLongTypeName *param2,
+		void *param3,
+		size_t param4,
+		const void *param5,
+		size_t param6)
+{
+	...
+}
+```
+
+No more than 7 parameters allowed (except for constructors).
+
+
+# Spaces
+
+* <span style="color: red">Do **not** use spaces to indent.</span>
```

Comment:
I would refrain from using such hacks in Markdown. It looks awkward if viewed in an editor. Does this really need to be red? Consider just emphasizing the line or the like.